x-pack/legacy/plugins/siem/public/components/open_timeline/delete_timeline_modal/index.tsx

```diff
@@ -16,23 +16,22 @@ interface Props {
   savedObjectId?: string | null;
   title?: string | null;
 }
-
-interface State {
-  showModal: boolean;
-}
-
 /**
  * Renders a button that when clicked, displays the `Delete Timeline` modal
  */
-export class DeleteTimelineModalButton extends React.PureComponent<Props, State> {
```

Comment:
nit - Nothing to do with your refactoring, but I am wondering if we can make it more precise than using `toggleShowModal` by `openModal` and `closeModal`